crypto-sodium/lib/Crypto/Hash.hs

```diff
@@ -0,0 +1,59 @@
+-- SPDX-FileCopyrightText: 2021 Serokell
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+{-# LANGUAGE ExplicitNamespaces, TypeOperators, TypeFamilies #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
+-- | Hashing
+module Crypto.Hash
+  ( blake2b
+  ) where
+
+import Prelude hiding (length)
+
+import Data.ByteArray (ByteArray, ByteArrayAccess, length, withByteArray)
+import Data.ByteArray.Sized (SizedByteArray, allocRet)
+import Data.Proxy (Proxy (Proxy))
+import Foreign.Ptr (nullPtr)
+import GHC.TypeNats (KnownNat, natVal, type (<=))
+
+import qualified Libsodium as Na
+
+-- | Hash returned by 'blake2b'.
+--
+-- This type is parametrised by hash size in bytes and the actual data type
+-- that contains bytes. This can be, for example, a @ByteString@.
+--
+-- Length must be between 16 and 64 bytes.
+type HashBlake2b len a = SizedByteArray len a
+
+-- | Hash a message using BLAKE2b
```

Comment:
Please align the Haddock with that of the SHA-2 functions. (Add a full stop at the end, add an example, etc.)